lib/report_portal/cucumber/report.rb

```diff
@@ -46,19 +47,45 @@ def initialize
       end
 
       def start_launch(desired_time = ReportPortal.now, cmd_args = ARGV)
+        # Not sure what is the use case if launch id is missing. But it does not make much of practical usage
+        #
+        # Expected behavior that make sense:
+        #  1. If launch_id present attach to existing (simple use case)
+        #  2. If launch_id not present check if exist rp_launch_id.tmp
+        #  3. [ADDED] If launch_id is not present check if lock exist with launch_uuid
         if attach_to_launch?
           ReportPortal.launch_id =
               if ReportPortal::Settings.instance.launch_id
                 ReportPortal::Settings.instance.launch_id
               else
-                file_path = ReportPortal::Settings.instance.file_with_launch_id || (Pathname(Dir.tmpdir) + 'rp_launch_id.tmp')
-                File.read(file_path)
+                file_path = lock_file
+                File.file?(file_path) ? read_lock_file(file_path) : new_launch(desired_time, cmd_args, file_path)
               end
           $stdout.puts "Attaching to launch #{ReportPortal.launch_id}"
         else
-          description = ReportPortal::Settings.instance.description
-          description ||= cmd_args.map {|arg| arg.gsub(/rp_uuid=.+/, "rp_uuid=[FILTERED]")}.join(' ')
-          ReportPortal.start_launch(description, time_to_send(desired_time))
+          new_launch(desired_time, cmd_args)
+        end
+      end
+
+      def new_launch(desired_time = ReportPortal.now, cmd_args = ARGV, lock_file = nil)
+        ReportPortal.start_launch(description(cmd_args), time_to_send(desired_time))
+        set_file_lock_with_launch_id(lock_file, ReportPortal.launch_id) if lock_file
+        ReportPortal.launch_id
+      end
+      
+      def description(cmd_args=ARGV)
+        description ||= ReportPortal::Settings.instance.description
+        description ||= cmd_args.map {|arg| arg.gsub(/rp_uuid=.+/, "rp_uuid=[FILTERED]")}.join(' ')
+        description
+      end
+
+      def set_file_lock_with_launch_id(lock_file, launch_id)
+        FileUtils.mkdir_p lock_file.dirname
```

Comment:
`lock_file` may return `ReportPortal::Settings.instance.file_with_launch_id`. Does it respond to ` dirname`?